src/meta/attribute.rs

```diff
@@ -1062,23 +1090,159 @@ impl ChannelList {
     }
 }
 
+// https://github.com/AcademySoftwareFoundation/openexr/blob/master/src/lib/OpenEXR/ImfTimeCode.cpp
 impl TimeCode {
 
     /// Number of bytes this would consume in an exr file.
     pub const BYTE_SIZE: usize = 2 * u32::BYTE_SIZE;
 
-    /// Without validation, write this instance to the byte stream.
+    /// Returns an error if this time code is considered invalid.
+    pub fn validate(&self, strict: bool) -> UnitResult {
+        if strict {
+            if self.frame > 29 { Err(Error::invalid("time code frame larger than 29")) }
+            else if self.seconds > 59 { Err(Error::invalid("time code seconds larger than 59")) }
+            else if self.minutes > 59 { Err(Error::invalid("time code minutes larger than 59")) }
+            else if self.hours > 23 { Err(Error::invalid("time code hours larger than 23")) }
+            else if self.binary_groups.iter().any(|&group| group > 4) {
+                Err(Error::invalid("time code binary group value too large for 3 bits"))
+            }
```

Comment:
Same here. I suspect the limit is 8: https://github.com/AcademySoftwareFoundation/openexr/blob/master/src/lib/OpenEXR/ImfTimeCode.cpp#L309

Reply:
I think the group parameter is an index, to identify one of 8 groups. 

The value uses three bits. So 2^3=8 should be the max. But when I tried to use 8 as max value, the set_bit function panicked because the value was too large, so I concluded that the limit must be 4. But it could also be 7, now that I think about it. 

Shouldn't 8 work for 3 bits?
1. `..000`
1. `..001` 
2. `..010`
3. `..011`
4. `..100`
5. `..101`
6. `..110`
7. `..111`
8. 
So 8 should not panic, right?

Reply:
15, because I had an indices bug, and this field actually has 4 bits available instead of only 3